GroupGame/Assets/Scripts/CharacterMove.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(CharacterController))]
+public class CharacterMove : MonoBehaviour
+{
+    public float speed = 3.0F;
+    public float rotateSpeed = 3.0F;
+
+    public float jumpSpeed = 8.0f;
+
+    public float gravity = 20.0f;
+    public Vector3 moveDirection = Vector3.zero;
+    
+    void Update()
+    {
+        CharacterController controller = GetComponent<CharacterController>();
+
+        // Rotate around y - axis
+        transform.Rotate(0, Input.GetAxis("Horizontal") * rotateSpeed, 0);
+
+        // Move forward / backward
+        Vector3 forward = transform.TransformDirection(Vector3.forward);
+        float curSpeed = speed * Input.GetAxis("Vertical");
+        controller.SimpleMove(forward * curSpeed);
+
+        if (Input.GetButtonDown ("Jump"))
+        {
+            if(controller.isGrounded)
+            { 
+                moveDirection.y= jumpSpeed;
+            }
```

Comment:
Something strange is happening here when jumping onto the trees. For a few frames after triggering a jump, if you collide with a surface then the jumpspeed is compounded onto the y velocity again and your jump height is increased. I'm not sure if you have much control over this since you're using a Unity Engine component but I think it may be worth taking a look at if this isn't expected behavior.

Additionally, you can't jump when traveling down the slopes you've created.

Reply:
I just resolved the jumping issue by implementing my own IsGrounded() method and also fixed an issue where the character is hovering above the ground.
I'm not sure what you mean by the jumpspeed compounding issue, but what you might be seeing is the fact that the trees just have a simple Capsule Collider, which means the top of the tree is rounded, not flat.